### DISCUSSION TITLE
SSE not working

### DISCUSSION THREAD

**Discussion by Tomgault1959** [mathieucarbou/ESPAsyncWebServer#166] [Q&A]
Hello, 

I decide to port all my projet from me-no-dev/ESPAsyncWebServer to mathieucarbou/ESPAsyncWebServer . All my SSE project work fine with me-no-dev/ESPAsyncWebServer but no one work with mathieucarbou/ESPAsyncWebServer. 

All project compile fine and start without error. ESP32 serial monitor log each  events.send but the browser receive nothing on console log. I tried with your example/sse_perftest (rework SSE handling) and nothing work.

Someone has a **working sketch** compatible with mathieucarbou/ESPAsyncWebServer to test it. 

Environment : 
Arduino IDE 2.3.4
ESP 3.0.7
mathieucarbou/AsyncTCP 3.2.14
mathieucarbou/ESPAsyncWebServer 3.3.23

I also tried with mathieucarbou/AsyncTCPSock without success

For example, the sketch https://randomnerdtutorials.com/esp32-web-server-sent-events-sse/ work fine with me-no-dev/ESPAsyncWebServer but not with mathieucarbou/ESPAsyncWebServer. Perhaps, i missed somethings when i read the documentation.

**Comment by mathieucarbou** [comment]
@vortigont : can you please check ?

**Comment by vortigont** [comment]
let me see

**Comment by vortigont** [comment]
ugh... looks like `String`s got me from behind :(
should fixed by #167 
@Tomgault1959 could you test with this?

**Reply by mathieucarbou** [reply]
> should fixed by #167

merged, thanks!

I've kept the SSE changes in main for now, not released, in order catch issues from people directly using the bleeding edge code from `main` branch.

**Reply by mathieucarbou** [reply]
@vortigont : I did a test, it works, and the perf tests shows a message rate between 450-480 messages per second with 10 concurrent workers.  This rocks!

```
Total: 1923 events, 480.75000000000000000000 events / second
Total: 1956 events, 489.00000000000000000000 events / second
Total: 1955 events, 488.75000000000000000000 events / second
Total: 1947 events, 486.75000000000000000000 events / second
Total: 1955 events, 488.75000000000000000000 events / second
Total: 1874 events, 468.50000000000000000000 events / second
Total: 1870 events, 467.50000000000000000000 events / second
Total: 1875 events, 468.75000000000000000000 events / second
Total: 1875 events, 468.75000000000000000000 events / second
Total: 1803 events, 450.75000000000000000000 events / second
```

**Reply by vortigont** [reply]
> I've kept the SSE changes in main for now, not released

agreed! let get some real feedback for this

**Comment by Tomgault1959** [comment]
Bonjour Mathieu, Hi Vortigont,

I recompile all my SSE project with 3.4.0 and now, all SSE events works !  The example/sse_perftest (rework SSE handling) also work.

Thank you very much Vortigont for your efficiency. Merci Mathieu pour la rapidité de la prise en charge du problème. 

Jean T Beaupré, Québec
